server/src/main/java/org/candlepin/service/impl/DefaultContentAccessCertServiceAdapter.java

```diff
@@ -0,0 +1,300 @@
+        StringBuffer contentPrefix = new StringBuffer();
+        contentPrefix.append("/");
+        contentPrefix.append(owner.getKey());
+        if (environment != null) {
+            contentPrefix.append("/");
+            contentPrefix.append(environment.getId());
+        }
+        return contentPrefix.toString();
+    }
+
+    private String createDN(Consumer consumer) {
+        StringBuilder sb = new StringBuilder("CN=");
+        sb.append(consumer.getId());
+        sb.append(", O=");
+        sb.append(consumer.getOwner().getId());
+        if (consumer.getEnvironment() != null) {
+            sb.append(", OU=");
+            sb.append(consumer.getEnvironment().getId());
+        }
+        return sb.toString();
+    }
+
+    public Set<X509ExtensionWrapper> prepareV3Extensions(Consumer consumer,
+        String contentPrefix, Map<String, EnvironmentContent> promotedContent) {
+        Set<X509ExtensionWrapper> result = v3extensionUtil.getExtensions(null,
+            contentPrefix, promotedContent);
+        return result;
+    }
+
+    public Set<X509ByteExtensionWrapper> prepareV3ByteExtensions(Consumer consumer,
+        org.candlepin.model.dto.Product container) throws IOException {
+        List<org.candlepin.model.dto.Product> products = new ArrayList<org.candlepin.model.dto.Product>();
+        products.add(container);
+        Set<X509ByteExtensionWrapper> result = v3extensionUtil.getByteExtensions(null,
+            products, null, null, null);
+        return result;
+    }
+
+    private byte[] createContentAccessDataPayload(Owner owner, Environment environment) throws IOException {
+        // fake a product dto as a container for the org content
+        Set<Product> containerSet = new HashSet<Product>();
+        List<Content> ownerContent = (List<Content>) ownerContentCurator.getContentByOwner(owner);
+        Set<String> entitledProductIds = new HashSet<String>();
+        List<org.candlepin.model.dto.Product> productModels =
+            new ArrayList<org.candlepin.model.dto.Product>();
+        Map<String, EnvironmentContent> promotedContent = getPromotedContent(environment);
+        String contentPrefix = getContentPrefix(owner, environment);
+        Product container = new Product();
+        Entitlement emptyEnt = new Entitlement();
+        Pool emptyPool = new Pool();
+        Product skuProduct = new Product();
+        Consumer emptyConsumer = new Consumer();
+
+        containerSet.add(container);
+        container.setId("content_access");
+        container.setName("Golden Ticket");
+        for (Content c : ownerContent) {
+            container.addContent(c, false);
+        }
+
+        emptyConsumer.setEnvironment(environment);
+        emptyEnt.setPool(emptyPool);
+        emptyEnt.setConsumer(emptyConsumer);
+        emptyPool.setProduct(skuProduct);
+        emptyPool.setStartDate(new Date());
+        emptyPool.setEndDate(new Date());
+        skuProduct.setName("Golden Ticket");
```

Comment:
perhaps "content access certificate" instead of Golden Ticket. As much as I love the name, exposing it to customers is likely to be confusing.